infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/K8sContainerResolver.java

```diff
@@ -100,4 +104,12 @@ private void provisionMemoryLimit(Container container, CheContainer cheContainer
         .map(e -> new io.fabric8.kubernetes.api.model.EnvVar(e.getName(), e.getValue(), null))
         .collect(Collectors.toList());
   }
+
+  private String buildContainerName(String pluginName, String cheContainerName) {
+    String preliminaryName = (pluginName + "-" + cheContainerName).toLowerCase();
+    if (preliminaryName.length() <= maxContainerNameLength) {
+      return preliminaryName;
+    }
+    return ("plugin-" + cheContainerName).substring(0, maxContainerNameLength - 1).toLowerCase();
```

Comment:
I think it makes sense to limit Che container name like to 50 characters and if `preliminaryName` is too long then get first 12 characters from Che Plugin Name concatenate Che container name along with '-' separator.

@garagatyi Please pay attention to this. Maybe you can suggest a better way.

Reply:
+